pkg/api/norman/customization/cluster/formatter.go

```diff
@@ -61,16 +45,43 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 	resource.Links["shell"] = shellLink
 	resource.AddAction(request, v32.ClusterActionGenerateKubeconfig)
 	resource.AddAction(request, v32.ClusterActionImportYaml)
+
+	// If user has permissions to update the cluster (regardless of RKE1 or not)
+	if canUpdateCluster(request) {
+		if convert.ToBool(resource.Values["enableClusterMonitoring"]) {
+			resource.AddAction(request, v32.ClusterActionDisableMonitoring)
+			resource.AddAction(request, v32.ClusterActionEditMonitoring)
+		} else {
+			resource.AddAction(request, v32.ClusterActionEnableMonitoring)
+		}
+	}
+
+	// If this is an RKE1 cluster only
 	if _, ok := resource.Values["rancherKubernetesEngineConfig"]; ok {
 		resource.AddAction(request, v32.ClusterActionExportYaml)
-		resource.AddAction(request, v32.ClusterActionRotateCertificates)
-		if rotateEncryptionKeyEnabled(f.clusterLister, resource.ID) {
-			resource.AddAction(request, v32.ClusterActionRotateEncryptionKey)
-		}
-		if _, ok := values.GetValue(resource.Values, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig"); ok {
+
+		// If a user has the BackupEtcd role/privilege, add it
+		if canBackupEtcd(request) {
 			resource.AddAction(request, v32.ClusterActionBackupEtcd)
-			resource.AddAction(request, v32.ClusterActionRestoreFromEtcdBackup)
 		}
+
+		// If user has permissions to update the cluster
+		if err := request.AccessControl.CanDo(v3.ClusterGroupVersionKind.Group, v3.ClusterResource.Name, "update", request, resource.Values, request.Schema); err == nil {
+			if _, ok := values.GetValue(resource.Values, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig"); ok {
```

Comment:
Can the signature of `canUpdateCluster` be modified so that it can be used here too?

Reply:
Totally agree, yep